lib/devise/models/confirmable.rb

```diff
@@ -169,6 +169,7 @@ def skip_reconfirmation!
         # in models to map to a nice sign up e-mail.
         def send_on_create_confirmation_instructions
           send_confirmation_instructions
+          skip_reconfirmation!
```

Comment:
Why don't you want add https://github.com/plataformatec/devise/blob/ac702843ddaa5652fde7a0050ff2378e7368149e/lib/devise/models/confirmable.rb#L46 to your model? It's hard to remove a callback in Rails, but still easy to add.